### ISSUE TITLE
Improve the Solution, using Directory.Build.props file

### ISSUE THREAD

**Issue by MithrilMan** [stratisproject/StratisBitcoinFullNode#3034]
I think we should fix the solution, compilation rises 5900+ warning and many of them are because we have projects that don't uses our ruleset.

An easy way is to use something I discovered few weeks ago, that's a special file where you declare default values for projects and I've used it in my own repo here: https://github.com/MithrilMan/StratisDebugFeature/blob/master/src/Directory.Build.props

basically in the src folder we place that file, that states default project configurations (in this case I specified LangVersion and analizer rules files, so we don't have to add them manually to the projects)

**Comment by MithrilMan** [comment]
I think this issue has already been solved by @bokobza, if he confirm then we can close this issue

**Comment by maciejzaleski** [comment]
@bokobza Could you please confirm if this has been already resolved.

**Comment by dangershony** [comment]
It was added here
https://github.com/stratisproject/StratisBitcoinFullNode/blob/master/src/Directory.Build.props
